### DISCUSSION TITLE
Boundary conditions that depend on gradients of the field

### DISCUSSION THREAD

**Discussion by JonasVeenstra** [zwicker-group/py-pde#439] [Q&A]
Dear Dr. Zwicker,

Thanks for developing an extremely useful package!

I am looking to implement boundary conditions that are functions of the field u **and its gradients**. 
I am aware that we can use bc={"value_expression": bc_value} to set the field and its derivative equal to a function of time, space and the fields.
Is it possible to construct a boundary condition that contains gradients and laplacians? 
For example, can I integrate the wave equation while constraining the laplacian of the field to be zero at the boundary?
![image](https://github.com/zwicker-group/py-pde/assets/93160674/6ffde044-8930-40d9-abf7-b6b48c27fb5c)

**Comment by david-zwicker** [comment]
I'm not a 100% sure what you're trying to achieve, but it is possible to control the second derivative in normal direction at the boundary using the `curvature` keyword; see [documentation of the boundary conditions](https://py-pde.readthedocs.io/en/latest/manual/advanced_usage.html#id1). You might thus achieve what you want by simply using `pde.WavePDE(bc={"curvature": 0})`.
